manifests/plugin/mysql/database.pp

```diff
@@ -2,8 +2,8 @@
 define collectd::plugin::mysql::database (
   Enum['present', 'absent'] $ensure    = 'present',
   String $database                     = $name,
-  Stdlib::Host $host                   = 'localhost',
```

Comment:
The change in the data type is OK though the default value should not change. This ensures it is backwards compatible while still allowing you to specify things differently for your use case.

Reply:
I don't think I understand; how can you allow undef *and* have a default value?

Reply:
A value with a default cannot be set to undef.

The underlaying question is: If I do not set a port, and if I set port to 3306, is the result the same?  For what I understand, the default `Port 3306` is now removed from the configuration, but if no port is the same as port 3306, it is not a breaking change because the bahaviour is the same.

If the behavior change (e.g. metrics are not collected, the name of the metric change, etc), then this is backward incompatible.

Reply:
For the port, you are correct; we could add back the default value there. For the host, however, the behavior is different...